### PR TITLE
[Release] Improve download interactions

### PR DESCRIPTION
**Pull request by turt2live**
https://github.com/matrix-org/matrix-react-sdk/pull/6509 to release

Conditional on design approval for the linked PR.

----

Notes: Improve the new download button's discoverability and interactions.

### CODE REVIEW

**Comment by turt2live** [comment]
I believe lint/test failures on this are because of the weird base branch.

File: src/utils/FileUtils.ts [hunk 1]
```diff
@@ -40,6 +42,21 @@ export function presentableTextForFile(
...
+    if (shortened && text.length > 19) {
+        const parts = text.split('.');
+        let fileName = parts.slice(0, parts.length - 1).join('.').substring(0, 15);
+        const extension = parts[parts.length - 1];
+
+        // Trim off any full stops from the file name to avoid a case where we
+        // add an ellipsis that looks really funky.
+        fileName = fileName.replace(/\.*$/g, '');
+
+        text = `${fileName}...${extension}`;
```
Comment (Palid) [inline src/utils/FileUtils.ts:57]: I think we should leave it to css if possible, this kind of trimming may turn out more destructive than helpful in some non-roman characters based languages.

**Review by Palid — APPROVED** [review]

File: src/utils/FileUtils.ts (hunk 1 above)
Comment (turt2live) [reply src/utils/FileUtils.ts:57]: I don't think css can trim in the middle of a word like this?

**Comment by dbkr** [comment]
nb. no need to put notes for a t-task since t-tasks won't be included anyway

**Pull request merged**